butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomChangePlugin.java

```diff
@@ -0,0 +1,252 @@
+
+    private String version;
+    private String extensions;
+    private List<PluginExecution> executions;
+    private List<Dependency> pluginDependencies;
+
+    // Removable properties, letting them to have default values, or be managed when applicable.
+    private boolean removeVersion = false;
+    private boolean removeExtensions = false;
+    private boolean removeExecutions = false;
+    private boolean removePluginDependencies = false;
+
+    // What to do if the dependency that is supposed to be changed is not present
+    private IfNotPresent ifNotPresent = IfNotPresent.Fail;
+
+    public PomChangePlugin() {
+    }
+
+    /**
+     * Operation to change a plugin in a Maven POM file.
+     * It allows changing version, extensions, executions and dependencies but not group id and artifact id.
+     * It also allows removing specific configuration, letting them
+     * to have default values, or be managed by applicable.
+     * <br>
+     * If the plugin to be changed doesn't actually exist, it will result
+     * in error
+     * <br>
+     * Important: no check will be done here for possible reasons to break
+     * the build, like the lack of version when the plugin is not managed
+     *
+     * @param groupId    plugin group id
+     * @param artifactId plugin artifact id
+     */
+    public PomChangePlugin(String groupId, String artifactId) {
+        setGroupId(groupId);
+        setArtifactId(artifactId);
+    }
+
+    /**
+     * Operation to set the version that will add/update the existing plugin version.
+     * @param   version Version to be set in the transformed plugin
+     * @return          this PomChangePlugin object
+     */
+    public PomChangePlugin setVersion(String version) {
+        checkForBlankString("Version", version);
+        this.version = version;
+        return this;
+    }
+
+    /**
+     * Operation to set the extensions that will add/update the existing plugin extensions.
+     * @param   extensions  Extension that needs to be set in the transformed plugin
+     * @return              this PomChangePlugin object
+     */
+    public PomChangePlugin setExtensions(String extensions) {
+        checkForBlankString("Extensions", extensions);
+        this.extensions = extensions;
+        return this;
+    }
+
+    /**
+     * Operation to set the Executions that will add/update the existing plugin executions.
+     * @param   executions  List of {@link PluginExecution} that needs to be set in the transformed plugin
+     * @return              this PomChangePlugin object
+     */
+    public PomChangePlugin setExecutions(List<PluginExecution> executions) {
+        checkForNull("Executions", executions);
+        Map<String, PluginExecution> executionMap = new LinkedHashMap<>();
+        for (PluginExecution exec : executions) {
+            if (executionMap.containsKey(exec.getId())) {
+                throw new IllegalStateException("You cannot have two plugin executions with the same " +
```

Comment:
Make this an `IllegalArgumentException` please

Reply:
Makes sense. Fixed this pushed code with `IllegalArgumentException`